modules/uncertainty/R/ensemble.R

```diff
@@ -200,35 +200,43 @@ get.ensemble.samples <- function(ensemble.size, pft.samples, env.samples,
 write.ensemble.configs <- function(defaults, ensemble.samples, settings, model, 
                                    clean = FALSE, write.to.db = TRUE,restart=NULL) {
   
+  con <- NULL
   my.write.config <- paste("write.config.", model, sep = "")
   my.write_restart <- paste0("write_restart.", model)
   
   if (is.null(ensemble.samples)) {
     return(list(runs = NULL, ensemble.id = NULL))
   }
   
-  # Open connection to database so we can store all run/ensemble information
+  # See if we need to write to DB
+  write.to.db <- as.logical(settings$database$bety$write)
+  
   if (write.to.db) {
-    con <- try(PEcAn.DB::db.open(settings$database$bety), silent = TRUE)
-    if (inherits(con, "try-error")) {
+    # Open connection to database so we can store all run/ensemble information
+    con <-
+      try(PEcAn.DB::db.open(settings$database$bety), silent = TRUE)
+    on.exit(try(PEcAn.DB::db.close(con), silent = TRUE)
+    )
```

Comment:
Why are you silencing errors here? At the very least, I would throw an informative warning, though I personally would be inclined to re-throw the error the same way you do in `read.write.configs`.

Reply:
Also, it's a good idea to call `on.exit` as `on.exit(..., add = TRUE)`. Otherwise, every invocation of `on.exit` will overwrite the previous one, which is usually not what we want.

From `?on.exit`:

```
     add: if TRUE, add ‘expr’ to be executed after any previously set
          expressions (or before if ‘after’ is FALSE); otherwise (the
          default) ‘expr’ will overwrite any previously set
          expressions.
```